### PR TITLE
[ktlint] Support `jvm_deps` and remove `cli`, `gradle`, `maven`

### LINKED ISSUE #1362: [ktlint] Support `jvm_deps` and reconsider schema
In #1000, `jvm_deps` was added to `sider.yml`.

But ktlint does not have it yet because ktlint has the complex schema (`cli`, `maven`, and `gradle`).
(see https://github.com/sider/runners/pull/1361#issuecomment-666869907 for details)

If we would support `jvm_deps`, I think these `cli`, `maven`, and `gradle` should be needless and removed.

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

For example:

```yaml
linter:
  ktlint:
    jvm_deps:
      - [foo, bar, 1.0.0]
    target:
      - "src/**/*.kt"
      - "!src/**/*Test.kt"
    ruleset:
      - path/to/my-ruleset.jar
    disabled_rules:
      - no-wildcard-imports
      - indent
      - experimental:indent
    experimental: true
```

> Link related issues or pull requests, e.g. `Fix #123`, `Related to #456`, or `None`.

Fix #1362

> Check the following items (please remove needless items).

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@yykamei This PR includes a breaking change. What do you think?

File: test/smokes/ktlint/expectations.rb [hunk 1]
```diff
@@ -147,13 +147,13 @@
 s.add_test(
   "maven",
```
Comment (yykamei) [inline test/smokes/ktlint/expectations.rb:148]: `App.kt` in the `maven` and the `gradle` directories seem to be already tested in the `success` directory. As far as I read the two directories, they would be no longer necessary.

```diff
--- ./gradle    2020-08-03 11:13:23.456601677 +0900
+++ ./success   2020-08-03 11:13:24.383314793 +0900
@@ -1,11 +1,22 @@
 s.add_test(
-  "gradle",
+  "success",
   type: "success",
   analyzer: { name: "ktlint", version: default_version },
   issues: [
     {
+      id: "18748d47ed6c20995492c1190b0fb829b794f8c1",
+      path: "src/Foo.kt",
+      location: { start_line: 1, start_column: 1 },
+      message: "Not a valid Kotlin file (1:1 expecting a top level declaration) (cannot be auto-corrected)",
+      links: [],
+      object: nil,
+      git_blame_info: {
+        commit: :_, line_hash: "cad1ff561173d394b00a49d1e560957407f7e67e", original_line: 1, final_line: 1
+      }
+    },
+    {
       id: "indent",
-      path: "src/main/kotlin/gradle/App.kt",
+      path: "src/App.kt",
       location: { start_line: 8, start_column: 1 },
       message: "Unexpected indentation (2) (should be 4)",
       links: [],
@@ -16,7 +27,7 @@
     },
     {
       id: "indent",
-      path: "src/main/kotlin/gradle/App.kt",
+      path: "src/App.kt",
       location: { start_line: 9, start_column: 1 },
       message: "Unexpected indentation (12) (should be 8)",
       links: [],
@@ -27,9 +38,9 @@
     },
     {
       id: "indent",
-      path: "src/main/kotlin/gradle/App.kt",
+      path: "src/App.kt",
       location: { start_line: 9, start_column: 34 },
-      message: 'Missing newline before "}"',
+      message: "Missing newline before \"}\"",
       links: [],
       object: nil,
       git_blame_info: {
```

```diff
--- ./maven     2020-08-03 11:15:06.721400787 +0900
+++ ./success   2020-08-03 11:13:24.383314793 +0900
@@ -1,22 +1,33 @@
 s.add_test(
-  "maven",
+  "success",
   type: "success",
   analyzer: { name: "ktlint", version: default_version },
   issues: [
     {
+      id: "18748d47ed6c20995492c1190b0fb829b794f8c1",
+      path: "src/Foo.kt",
+      location: { start_line: 1, start_column: 1 },
+      message: "Not a valid Kotlin file (1:1 expecting a top level declaration) (cannot be auto-corrected)",
+      links: [],
+      object: nil,
+      git_blame_info: {
+        commit: :_, line_hash: "cad1ff561173d394b00a49d1e560957407f7e67e", original_line: 1, final_line: 1
+      }
+    },
+    {
       id: "indent",
-      path: "src/main/App.kt",
+      path: "src/App.kt",
       location: { start_line: 8, start_column: 1 },
       message: "Unexpected indentation (2) (should be 4)",
       links: [],
       object: nil,
       git_blame_info: {
-        commit: :_, line_hash: "a7efb26ba5408fc562917a1e5f23e6413ce55af2", original_line: 8, final_line: 8
+        commit: :_, line_hash: "6619184f836068051c8c20ecda475e7f8c99ca76", original_line: 8, final_line: 8
       }
     },
     {
       id: "indent",
-      path: "src/main/App.kt",
+      path: "src/App.kt",
       location: { start_line: 9, start_column: 1 },
       message: "Unexpected indentation (12) (should be 8)",
       links: [],
@@ -27,9 +38,9 @@
     },
     {
       id: "indent",
-      path: "src/main/App.kt",
+      path: "src/App.kt",
       location: { start_line: 9, start_column: 34 },
-      message: 'Missing newline before "}"',
+      message: "Missing newline before \"}\"",
       links: [],
       object: nil,
       git_blame_info: {
```

Comment (ybiquitous) [reply test/smokes/ktlint/expectations.rb:148]: You are right. I will reduce the test cases. 👍

Comment (ybiquitous) [reply test/smokes/ktlint/expectations.rb]: Fixed via b248d841

**Pull request merged**